src/playlist-maker.js

```diff
@@ -22,144 +30,175 @@ var indexInSources = function(arr, src) {
   return -1;
 };
 
-// factory method to return a new playlist with the following API
-// playlist(["a", "b", "c"]) // setter, ["a", "b", "c"]
-// playlist() // getter, ["a", "b", "c"]
-// playlist.currentItem() // getter, 0
-// playlist.currentItem(1) // setter, 1
-// playlist.next() // "c"
-// playlist.previous() // "b"
-var playlistMaker = function(player, plist) {
-  var currentIndex = -1;
-  var autoadvanceTimeout = null;
-  var list = [];
-  var playlistchangeTimeout;
-  var loadFirstItem = function loadFirstItem() {
-    if (list.length > 0) {
-      currentIndex = 0;
-      playItem(player, autoadvanceTimeout, list[0]);
-    } else {
-      currentIndex = -1;
-    }
-  };
-
-  if (plist && isArray(plist)) {
-    list = plist.slice();
-  }
-
-  player.on('dispose', function() {
-    window.clearTimeout(playlistchangeTimeout);
-    playlistchangeTimeout = null;
-  });
-
-  var playlist = function playlist(plist) {
-    if (plist && isArray(plist)) {
-      list = plist.slice();
-      loadFirstItem();
-
-      playlistchangeTimeout = window.setTimeout(function() {
+/**
+ * Factory function for creating new playlists on the given player.
+ *
+ * @param  {Player} player
+ * @param  {Array} initialList
+ * @return {[type]}
+ */
+const factory = (player, initialList) => {
+
+  /**
+   * Get/set the playlist for a player.
+   *
+   * @param  {Array} [list]
+   * @return {Array}
+   */
+  const playlist = player.playlist = function(list) {
+    if (Array.isArray(list)) {
+      playlist.list_ = list.slice();
```

Comment:
Yep, only used inside the `factory` method. The rest of the variables are the same. Though, `list_` is the main one I *really* care about.